CesiumUtility/include/CesiumUtility/JsonValue.h

```diff
@@ -620,6 +620,48 @@ class CESIUMUTILITY_API JsonValue final {
     return std::holds_alternative<std::int64_t>(this->value);
   }
 
+  /**
+   * @brief Returns the size in bytes of this `JsonValue`.
+   */
+  int64_t getSizeBytes() const noexcept {
+    struct Operation {
+      int64_t operator()([[maybe_unused]] const Null& value) {
+        return sizeof(std::nullptr_t);
+      }
+      int64_t operator()([[maybe_unused]] const double& value) {
+        return sizeof(double);
+      }
+      int64_t operator()([[maybe_unused]] const std::uint64_t& value) {
+        return sizeof(std::uint64_t);
+      }
+      int64_t operator()([[maybe_unused]] const std::int64_t& value) {
+        return sizeof(std::int64_t);
+      }
+      int64_t operator()([[maybe_unused]] const Bool& value) {
+        return sizeof(Bool);
+      }
+      int64_t operator()(const String& value) { return value.size(); }
+      int64_t operator()(const Object& value) {
+        int64_t accum = 0;
+        for (auto& [k, v] : value) {
+          accum += k.size();
+          accum += v.getSizeBytes();
+        }
+
+        return accum;
+      }
+      int64_t operator()(const Array& value) {
+        int64_t accum = 0;
+        for (const JsonValue& v : value) {
+          accum += v.getSizeBytes();
+        }
+        return accum;
+      }
+    };
+
+    return std::visit(Operation{}, this->value);
```

Comment:
The size should always be `sizeof(JsonValue)` plus some extra externally-allocated memory for strings, arrays, and objects. The logic for the extra should be the same as for fields of those types in regular objects.